java/code/src/com/redhat/rhn/common/security/acl/Access.java

```diff
@@ -100,6 +102,41 @@ public boolean aclUserRole(Map<String, Object> ctx, String[] params) {
         return false;
     }
 
+    /**
+     * Returns <i>true</i> if the current user is permitted access to a namespace<strong>or any of its
+     * parents</strong>.
+     * <p>
+     * This method is not suitable for access filtering and should be preferred only for conditional rendering
+     * as it returns also <i>true</i> if the user has access to any parent of the specified namespace.
+     * @param ctx the context map
+     * @param params the namespace and the access mode ('R' for read-only, 'W' for write)
+     *               as respective elements of an array
+     * @return <i>true</i> if user is permitted access to the specified namespace or any of its parents
+     */
+    public boolean aclAuthorizedFor(Map<String, Object> ctx, String[] params) {
+        User user = (User) ctx.get("user");
+        if (user == null) {
+            return false;
+        }
+
+        if (user.hasRole(RoleFactory.SAT_ADMIN)) {
+            return true;
+        }
+
+        boolean authorized = UserManager.getPermittedNamespaces(user)
+                .filter(ns -> params[0].equals(ns.getNamespace()) ||
+                        ns.getNamespace().startsWith(params[0] + '.')
+                ).anyMatch(ns -> params.length < 2 ||
```

Comment:
we should have some comments explaining this magic number. Why 2? (I'm assuming that it must have and least one parent, otherwise it would match the full path)

Reply:
Right, this one needs a description.

Second element in params is the optional "access mode". If not provided, either R or W would match.

Reply:
Here are some examples: https://github.com/uyuni-project/uyuni/commit/4f50a5d6ac351a88bc6226e425b5d0a42deb86cc